finality-aleph/src/justification.rs

```diff
@@ -64,111 +85,114 @@ where
     pub number: NumberFor<Block>,
 }
 
-pub(crate) struct JustificationHandler<B, RB, C, BE>
+pub(crate) struct JustificationHandler<B, RB, C, BE, D, SI, F>
 where
     B: BlockT,
     RB: network::RequestBlocks<B> + 'static,
     C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
```

Comment:
I think it might be a good step, to completely remove ClientForAleph as a set of traits. For example this line could be changed to:
```
    C: HeaderBackend<B> + LockImportRun<B, BE> + Finalizer<B, BE> + Send + Sync + 'static,
```
as in JustificationHandler we only use `info`, `header` and we call `finalizer::finalize_block`, which uses `lock_import_and_run`, `apply_finality` and `info`. We don't need other traits here. This way mocking client would be easier and maybe we could mock `block_requester` too?

Reply:
good catch, thanks! I have minimized the required trait; I think that `RequestBlocks` is relatively easy to mock